src/app/cli/src/coda.ml

```diff
@@ -133,284 +133,312 @@ let daemon logger =
          else Sys.home_directory () >>| compute_conf_dir
        in
        Parallel.init_master () ;
-       let%bind config =
-         match%map
-           Monitor.try_with_or_error ~extract_exn:true (fun () ->
-               let%bind r = Reader.open_file (conf_dir ^/ "daemon.json") in
-               let%map contents =
-                 Pipe.to_list (Reader.lines r)
-                 >>| fun ss -> String.concat ~sep:"\n" ss
+       let monitor = Async.Monitor.create ~name:"coda" () in
+       let module Coda_initialization = struct
+         type ('a, 'b, 'c, 'd, 'e) t =
+           { coda: 'a
+           ; client_whitelist: 'b
+           ; rest_server_port: 'c
+           ; client_port: 'd
+           ; run_snark_worker_action: 'e }
+       end in
+       let coda_initialization_deferred =
```

Comment:
This doesn't prevent the code inside the deferred from running. You should make this a `let coda_initialized_deferred () = ` to have the effect you want.

Reply:
Actually I'm immediately starting the deferred there on purpose, but perhaps its cleaner to start it closer (in the sense of the line of code) to where it's being used

Reply:
By starting it here, you aren't running it inside the monitor.